visualizer/src/service/service.js

```diff
@@ -1,13 +1,49 @@
 import { interpret } from 'xstate';
 import createProjectMachine from './projectMachine';
+import createQualityControlMachine from './qualityControlMachine';
 
-const location = window.location;
-const search = new URLSearchParams(location.search);
-const projectId = search.get('projectId');
-const bucket = search.has('bucket') ? search.get('bucket') : 'caliban-output';
-const machine = createProjectMachine(projectId, bucket);
-const service = interpret(machine); // , { devTools: true });
-service.start();
-window.dcl = service;
+export function isProjectId(id) {
+  // Checks id is a 12 character URL-safe base64 string
+  // URL-safe base 64 uses - instead of + and _ instead of /
+  const projectIdRegex = /^[\w\-]{12}$/;
+  return projectIdRegex.test(id);
+}
 
-export default service;
+export function isQualityControl(ids) {
+  // Checks ids is a comma separated list of project IDs
+  return ids.split(',').every(isProjectId);
+}
+
+function createProject() {
+  const location = window.location;
+  const search = new URLSearchParams(location.search);
+  const projectId = search.get('projectId');
+  if (!projectId || projectId.split(',').length > 1) {
+    return;
+  }
+  const bucket = search.has('bucket') ? search.get('bucket') : 'caliban-output';
+  const machine = createProjectMachine(projectId, bucket);
+  const service = interpret(machine); // , { devTools: true });
```

Comment:
Minor: Does React have a debug vs production mode? Maybe we can leave the `devtools` in and set the value to be "isDebugMode"?

Reply:
We could use `process.env.NODE_ENV` to distinguish between dev and prod

> There is also a built-in environment variable called NODE_ENV. You can read it from process.env.NODE_ENV. When you run npm start, it is always equal to 'development', when you run npm test it is always equal to 'test', and when you run npm run build to make a production bundle, it is always equal to 'production'. You cannot override NODE_ENV manually. 

(from [https://create-react-app.dev/docs/adding-custom-environment-variables/](https://create-react-app.dev/docs/adding-custom-environment-variables/))

It looks like we can't add a debug mode on top of these, and I often do not want the devtools enabled as they can slow things down significantly. I uncomment the devtools occasionally to visual inspect machines, but I think leaving it commented out is correct so that it's not unintentionally enabled by new developers.